### ISSUE TITLE
teleporter doesn't save all settings

### ISSUE THREAD

**Issue by PA3DSS** [pi-hole/pi-hole#4290]
### Versions

Pi-hole version is v5.3.1 (Latest: v5.3.1)
AdminLTE version is v5.5.1 (Latest: v5.5.1)
FTL version is v5.8.1 (Latest: v5.8.1)

### Platform
Raspberry pi with Buster include all updates.

It is not really a bug but teleporter doesn't save these 2 settings i noticed
![image](https://user-images.githubusercontent.com/46996977/130819614-2abaf437-25e5-4c80-b2e8-07513b913c04.png)

![image](https://user-images.githubusercontent.com/46996977/130819699-f68b2527-e717-4f09-b6c1-1dcc1e114c59.png)

**Comment by yubiuser** [comment]
Actually, all these options **are exported but not imported** by teleporter. The options are saved to `/etc/pihole/setupVars.conf` which is included in the *.zip file. However, it is not restored on import.

I just wonder if the behavior should be changed, as Pi-hole so not going to use the information anymore:
https://github.com/pi-hole/FTL/pull/1111
https://github.com/pi-hole/pi-hole/pull/4246

**Comment by dschaper** [comment]
If it's a bug that needs to be fixed then we should fix it and not wait until a later date where things may be different.

**Comment by yubiuser** [comment]
@DL6ER 

You wrote the initial teleporter function (https://github.com/pi-hole/AdminLTE/pull/369). It already contained `setupVars.conf`, but only exported it, but never imported it. There was/is also no checkbox for it. Was there a reason behind it? Or was it simply forgotten?

**Comment by DL6ER** [comment]
`setupVars.conf` is/was intentionally not imported because an import of an exported teleporter file from a *different Pi-hole could have lead to severe issues due to the incorrect `IPV{4,6}_ADDR`. Now that the IP address is gone, we could add importing `setupVars.conf`.

Pi-hole v6.0 is going to make a lot of things different, including not using `setupVars.conf` for the web interface. However, for the time being, importing `setupVars.conf` should be safe.

**Comment by ramdyne** [comment]
I am sure the devs know this, but for users like me: things like DHCP server settings are not restored. This is preventing me from migrating my current install to a different setup which is really a bummer.

**Comment by unikitty37** [comment]
Yes, my DHCP server settings were not restored. Additionally, the old system running on a now-unsupported version of Raspbian correctly provided DNS entries for devices given addresses by local DHCP, so if DHCP assigned a device with hostname `sausage` to 192.168.1.42, then `dig sausage` would return 192.168.1.42. Now I just get NXDOMAIN.

How, short of screenshotting *every* page on the old system and painstakingly cross-checking *every* page on the new one, can I get *every* setting transferred?

Surely the logic should be "if a setting exists on the system, put it in the backup. If a setting exists in the backup, restore it to the system"?

I'm going back to the un-upgradeable Raspbian 9 system. At least it *works* for its intended purpose.

**Comment by yubiuser** [comment]
If you used static DHCP assignments for your devices you should find `"04-pihole-static-dhcp.conf` in your teleporter backup archive. Just copy this to `/etc/dnsmasq.d/`

**Comment by unikitty37** [comment]
Sorry, I deleted the original comment because it felt a fair bit too ranty.

The static DHCP assignments were copied across — I could see them in the UI. But DNS lookups were failing. I hadn't manually copied `setupVars.conf`, which would have helped with the missing DHCP settings, but I can't really see anything in there that would have broken the DNS.

When I've got time to restart the Pihole, I'll try manually copying that across and seeing if it fixes it.

**Comment by unikitty37** [comment]
Copying `setupVars.conf` across sort of fixed it, but DNS entries only showed for entries in the new server's lease table; most devices weren't resolvable until their leases expired or were force-renewed, meaning DNS was wonky for 24 hours.

I'm guessing the leases aren't stored in a file that can be simply copied across easily, though. Would definitely prefer it if static DHCP leases could be put into DNS without having had to see the device first, but that's for another ticket…

**Comment by yubiuser** [comment]
A meantioned above Pi-hole v6 will handle Teleporter via ``FTL``. It does export/restore DNSSEC and theme settings/boxed layout

**Comment by yubiuser** [comment]
Pi-hole v6 has been released.

**Comment by thomasmerz** [comment]
…but where has `teleporter` CLI gone in V6? 🤷‍♂  `pihole -a teleporter` as before isn't a valid command anymore?!

**Comment by jfb-pihole** [comment]
See `pihole-FTL --help`

The function has moved to FTL.
